lib/cloudshaper/stack_module.rb

```diff
@@ -90,9 +90,15 @@ def register_resource(resource_type, name, &block)
     end
 
     def register_variable(name, &block)
+      return if @stack_elements[:variable].key?(name)
```

Comment:
Silently, at least right now, because variables may have already been defined before module build when being initialized from `stacks.yaml` (which isn't an error). If anything, we can shuffle things around to build and then initialize from `stacks.yaml`, but then we would silently overwrite.